tests/jballerina-unit-test/src/test/resources/test-src/statements/matchstmt/varbindingpatternmatchpattern/mapping_binding_pattern.bal

```diff
@@ -365,6 +365,64 @@ function testMappingBindingPattern17() {
     assertEquals((), mappingBindingPattern17(1));
 }
 
+type UnaryExpr record {|
+    string operand;
+|};
+
+type TypeCastExpr record {|
+    string oprd;
+|};
+
+type Expr UnaryExpr|TypeCastExpr;
+
+function testTypeNarrowingInMatchStmt1(Expr expr) {
+    match expr {
+        var {oprd: o} => {
+            TypeCastExpr c = expr;
+            assertEquals("sample", c.oprd);
+        }
+    }
+}
+
+function testTypeNarrowingInMatchStmt2(Expr expr) {
+    match expr {
+        var {operand : o} => {
+            UnaryExpr c = expr;
+        }
+        var {oprd: o} => {
+            TypeCastExpr c = expr;
+            assertEquals("sample", c.oprd);
+        }
+    }
+}
+
+function testTypeNarrowingInMatchStmt3(Expr expr) {
+    Expr x = {oprd : "sample"};
+    match x {
+        {oprd: "sample"} => {
+            TypeCastExpr c = x;
+            assertEquals("sample", c.oprd);
+        }
+    }
+}
+
+function testTypeNarrowingInMatchStmt4(Expr expr) {
+    match expr {
+        {oprd: var o} => {
+            TypeCastExpr c = expr;
+            assertEquals("sample", c.oprd);
+        }
+    }
+}
```

Comment:
Should we add tests cases with `const-match` and `list-match` pattern as well?

Reply:
@rdulmina const-match is already tested with `testTypeNarrowingInMatchStmt2` I will add a test for list-match

Reply:
Okay cool